staging_vespalib/src/vespa/vespalib/util/singleexecutor.cpp

```diff
@@ -41,12 +41,38 @@ SingleExecutor::addTask(Task::UP task) {
     return wp;
 }
 
+void
+SingleExecutor::wakeupConsumer() {
+    MonitorGuard guard(_consumerMonitor);
+    guard.signal();
+}
+
+void
+SingleExecutor::sleepConsumer() {
+    _wakeupConsumerAt.store(_wp.load(std::memory_order_relaxed) + (_taskLimit.load(std::memory_order_relaxed) >> 2), std::memory_order_relaxed);
+    MonitorGuard guard(_consumerMonitor);
+    guard.wait(10ms);
+    _wakeupConsumerAt.store(0, std::memory_order_relaxed);
+}
+
+void
+SingleExecutor::wakeupProducer() {
+    MonitorGuard guard(_producerMonitor);
```

Comment:
You can skip taking the lock here; consider using std::mutex and std::condition_variable instead.

Reply:
Done